File: src/main/c-cpp/personal/Factorial.cpp
#include <iostream>

using namespace std;

int factorial(int number)
{
	if(number == 1 || number == 0)
	{
		return number;
	}
	else
		return number * factorial(number - 1);
}

int main(int argc, char* argv[])
{
	int number = 5;
	cout<<"Factorial of "<<number<<" is "<<factorial(number)<<endl;
}

File: src/main/java/com/utility/Factorial.java
package com.utility;

public class Factorial {

    public static long getFactorial(long number) {
        long factorial = 1;

        for (long i = 1; i <= number; i++) {
            factorial = i * factorial;
        }
        return factorial;
    }

    public static void main(String[] args) {

        System.out.println(getFactorial(15));
    }

}
